lib/ansible/modules/windows/win_auto_logon.ps1

```diff
@@ -9,75 +9,75 @@
     }
-    $actionTaken = $null
-    $autoLogonRegPath   = 'HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Winlogon\'
-    $autoLogonKeyRegList   = Get-ItemProperty -LiteralPath $autoLogonRegPath -Name $autoLogonKeyList.GetEnumerator().Name -ErrorAction SilentlyContinue
+    $ntAccount = $accountSid.Translate([System.Security.Principal.NTAccount])
+
+    $domain, $username = $ntAccount.Value -split '\\'
+}
 
-    Foreach($key in $autoLogonKeyList.GetEnumerator().Name){
-        $currentKeyValue = $autoLogonKeyRegList | Select-Object -ExpandProperty $key -ErrorAction SilentlyContinue
-        if (-not [String]::IsNullOrEmpty($currentKeyValue)) {
-            $expectedValue = $autoLogonKeyList[$key]
-            if(($state -eq 'present') -and ($currentKeyValue -ne $expectedValue)) {
-                Set-ItemProperty -LiteralPath $autoLogonRegPath -Name $key -Value $autoLogonKeyList[$key] -Force
-                $actionTaken = $true
-            }
-            elseif($state -eq 'absent') {
-                $actionTaken = $true
-                Remove-ItemProperty -LiteralPath  $autoLogonRegPath -Name $key -Force
-            }
+#Build ParamHash
+
+$autoAdminLogon = 1
+if($state -eq 'absent'){
+    $autoadminlogon = 0
+}
+$autoLogonKeyList   = @{
+    DefaultPassword = $password
+    DefaultUserName = $username
+    DefaultDomain   = $domain
+    AutoAdminLogon  = $autoAdminLogon
+}
+$actionTaken = $null
+$autoLogonRegPath   = 'HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Winlogon\'
+$autoLogonKeyRegList   = Get-ItemProperty -LiteralPath $autoLogonRegPath -Name $autoLogonKeyList.GetEnumerator().Name -ErrorAction SilentlyContinue
+
+Foreach($key in $autoLogonKeyList.GetEnumerator().Name){
+    $currentKeyValue = $autoLogonKeyRegList | Select-Object -ExpandProperty $key -ErrorAction SilentlyContinue
+    if (-not [String]::IsNullOrEmpty($currentKeyValue)) {
+        $expectedValue = $autoLogonKeyList[$key]
+        if(($state -eq 'present') -and ($currentKeyValue -ne $expectedValue)) {
+            Set-ItemProperty -LiteralPath $autoLogonRegPath -Name $key -Value $autoLogonKeyList[$key] -Force
+            $actionTaken = $true
         }
-        else {
-            if ($state -eq 'present') {
-                $actionTaken = $true
-                New-ItemProperty -LiteralPath $autoLogonRegPath -Name $key -Value $autoLogonKeyList[$key] -Force | Out-Null
-            }
+        elseif($state -eq 'absent') {
+            $actionTaken = $true
+            Remove-ItemProperty -LiteralPath  $autoLogonRegPath -Name $key -Force
         }
     }
-    if($actionTaken){
-        $module.Result.Msg     = "Auto logon registry keys are $stateMessage"
-        $module.Result.changed = $true
+    else {
+        if ($state -eq 'present') {
+            $actionTaken = $true
+            New-ItemProperty -LiteralPath $autoLogonRegPath -Name $key -Value $autoLogonKeyList[$key] -Force | Out-Null
+        }
     }
 }
-catch {
```

Comment:
Avoid the generic `try/catch` over large blocks of code. Since Ansible 2.8 we will produce a fairly robust error message for uncaught exceptions and doing this actually removes some useful information like the stack trace and PSError message itself.

Only use try/catch if there is a specific exception that you can recover from or the actual exception raised is too vague and we can provide better context around it.